test-result-summary-client/src/Build/ReleaseSummary.jsx

```diff
@@ -60,25 +61,52 @@ export default class ReleaseSummary extends Component {
                                 : ` ❌ ${buildResult} ❌${nl}`;
 
                         if (buildName.startsWith('Test_openjdk')) {
-                            const rerunLinkInfo = rerunLink
-                                ? `Rerun [all](${rerunLink}) ${nl}`
+                            tests.forEach((test) => {
+                                if (test.testResult === 'FAILED') {
+                                    if (
+                                        !this.state.failedTests.includes(
+                                            test.testName
+                                        )
+                                    ) {
+                                        this.state.failedTests.push(
+                                            test.testName
+                                        );
+                                    }
+                                }
+                            });
+                            const rerunFailedLink = rerunLink
+                                ? rerunLink.replace(
+                                      /(\WTARGET=)([^&]*)/gi,
+                                      '$1' + this.state.failedTests
+                                  )
```

Comment:
I'm grabbing the `rerunLink` from the console output for the rerun all in grinder link. Whereas with the rerun failed tests link, should I instead be grabbing the `rerunLink` from the testList builds? For example, in the console output of a build, i.e.
```_id	"6248f39a72761a4c5cfdc804"
url	"https://ci.adoptopenjdk.net"
buildName	"Test_openjdk11_j9_extended.openjdk_s390x_linux_testList_1"
buildNameStr	"Test_openjdk11_j9_extended.openjdk_s390x_linux_testList_1"
buildNum	31
rootBuildId	"624883e772761a4c5cfc1cf5"
parentId	"6248f28872761a4c5cfdc3e4"
type	"Test"
status	"Done"
artifactory	null
buildDuration	7949018
buildParams	[ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
buildResult	"UNSTABLE"
buildUrl	"https://ci.adoptopenjdk.net/job/Test_openjdk11_j9_extended.openjdk_s390x_linux_testList_1/31/"
hasChildren	false
javaVersion	"openjdk version \"11.0.15\" 2022-04-19\r\nOpenJDK Runtime Environment (build 11.0.15+8)\r\nEclipse OpenJ9 VM (build master-399178567, JRE 11 Linux s390x-64-Bit Compressed References 20220402_1098 (JIT enabled, AOT enabled)\r\nOpenJ9 - 399178567\r\nOMR - 923ae683b\r\nJCL - d112e5d42f based on jdk-11.0.15+8)"
jdkDate	" 2022-04-19"
machine	"test-marist-ubuntu1804-s390x-2"
parserType	"Test"
rerunLink	"https://ci.adoptopenjdk.net/job/Grinder/parambuild/?SDK_RESOURCE=upstream&amp;TARGET=-f+parallelList.mk+testList_1&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk11_j9_extended.openjdk_s390x_linux&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=AdoptOpenJDK%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=s390x_linux&amp;GENERATE_JOBS=false&amp;KEEP_REPORTDIR=true&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_…;DOCKERIMAGE_TAG=&amp;JDK_IMPL=openj9&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=&amp;DYNAMIC_COMPILE=false&amp;TKG_SHA=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;TRSS_URL=&amp;USE_TESTENV_PROPERTIES=false&amp;BUILD_LIST=openjdk&amp;UPSTREAM_JOB_NUMBER=1098&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=10&amp;JVM_OPTIONS=&amp;PARALLEL=None"
sdkResource	"upstream"
startBy	"upstream project \"Test_openjdk11_j9_extended.openjdk_s390x_linux\" build number 58"
testSummary	Object { total: 78, executed: 8, passed: 6, … }
tests	[ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
timestamp	1648939805642
versions	Object { }
```

Should I just grab `rerunLink` for the rerun failted tests in grinder? If so, wouldn't that mean I would have to combine all the subsidary testLists' targets from each rerunLink so that it would stitch to something like
```testList TESTLIST=testList_1,testList_2``` ?
I'm not seeing an example where the rerunLink will provide me with the target of all failed test lists, so I'm unsure if I have to grab them manually through the buildNames of each testLists

Reply:
The above is not the Jenkins console output of a build. It is data from our database.
As mentioned, the information is not in the database. We need to add logic to parse the Jenkins console output and store the data in the database on the server-side. Not on the client-side (above). 

For example Jenkins console output:
https://ci.adoptopenjdk.net/job/Test_openjdk18_hs_extended.openjdk_x86-64_linux/23/console

```
Rerun in Grinder with failed test targets: [https://ci.adoptopenjdk.net/job/Grinder/parambuild/?SDK_RESOURCE=&amp;TARGET=testList+TESTLIST=jdk_net_0,jdk_jfr_1,jdk_management_1,jdk_nio_0,jdk_management_0&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_linux&amp;GENERATE_JOBS=false&amp;KEEP_REPORTDIR=true&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=&amp;BUILD_IDENTIFIER=&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=&amp;JDK_VERSION=18&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Fjdk18u&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;UPSTREAM_JOB_NAME=build-scripts%2Fjobs%2Fjdk18u%2Fjdk18u-linux-x64-temurin&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;CLOUD_PROVIDER=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=dev&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=&amp;DOCKERIMAGE_TAG=&amp;JDK_IMPL=hotspot&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=&amp;DYNAMIC_COMPILE=false&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=3&amp;OPENJDK_SHA=&amp;TRSS_URL=&amp;USE_TESTENV_PROPERTIES=false&amp;BUILD_LIST=openjdk&amp;UPSTREAM_JOB_NUMBER=12&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=10&amp;JVM_OPTIONS=&amp;PARALLEL=Dynamic](https://ci.adoptopenjdk.net/job/Grinder/parambuild/?SDK_RESOURCE=&TARGET=testList+TESTLIST=jdk_net_0,jdk_jfr_1,jdk_management_1,jdk_nio_0,jdk_management_0&TEST_FLAG=&UPSTREAM_TEST_JOB_NAME=&DOCKER_REQUIRED=false&ACTIVE_NODE_TIMEOUT=0&VENDOR_TEST_DIRS=&EXTRA_DOCKER_ARGS=&TKG_OWNER_BRANCH=adoptium%3Amaster&OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&PLATFORM=x86-64_linux&GENERATE_JOBS=false&KEEP_REPORTDIR=true&PERSONAL_BUILD=false&ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&LABEL=&EXTRA_OPTIONS=&CUSTOMIZED_SDK_URL=&BUILD_IDENTIFIER=&ADOPTOPENJDK_BRANCH=master&LIGHT_WEIGHT_CHECKOUT=false&USE_JRE=false&ARTIFACTORY_SERVER=&KEEP_WORKSPACE=false&USER_CREDENTIALS_ID=&JDK_VERSION=18&ITERATIONS=1&VENDOR_TEST_REPOS=&JDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Fjdk18u&OPENJ9_BRANCH=master&OPENJ9_SHA=&JCK_GIT_REPO=&VENDOR_TEST_BRANCHES=&UPSTREAM_JOB_NAME=build-scripts%2Fjobs%2Fjdk18u%2Fjdk18u-linux-x64-temurin&OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&CLOUD_PROVIDER=&CUSTOM_TARGET=&VENDOR_TEST_SHAS=&JDK_BRANCH=dev&LABEL_ADDITION=&ARTIFACTORY_REPO=&ARTIFACTORY_ROOT_DIR=&UPSTREAM_TEST_JOB_NUMBER=&DOCKERIMAGE_TAG=&JDK_IMPL=hotspot&TEST_TIME=&SSH_AGENT_CREDENTIAL=&AUTO_DETECT=true&SLACK_CHANNEL=&DYNAMIC_COMPILE=false&ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=&ARCHIVE_TEST_RESULTS=false&NUM_MACHINES=3&OPENJDK_SHA=&TRSS_URL=&USE_TESTENV_PROPERTIES=false&BUILD_LIST=openjdk&UPSTREAM_JOB_NUMBER=12&STF_OWNER_BRANCH=adoptium%3Amaster&TIME_LIMIT=10&JVM_OPTIONS=&PARALLEL=Dynamic)
```